### DISCUSSION TITLE
Netbox Device Roles Issue

### DISCUSSION THREAD

**Discussion by ds2600** [peering-manager/peering-manager#656] [Q&A] [answered]
Testing out Peering Manager in a dev/Docker environment to verify utility for staff. Netbox info from peering-manager.env is:
 ```
NETBOX_API="http://192.168.1.112"
NETBOX_API_TOKEN=abcdefgh123456
NETBOX_API_THREADING=False
NETBOX_API_VERIFY_SSL=False
NETBOX_DEVICE_ROLES=["edge-router"]
 ```
When attempting to add a router in Peering Manager I get the following error:
 ```
The request failed with code 400 Bad Request: {'role': ['Select a valid choice. ["edge-router"] is not one of the available choices.']}
 ```

However, this device role is definitely available in Netbox:
 ```
{
            "id": 1,
            "url": "http://192.168.1.112/api/dcim/device-roles/1/",
            "display": "Edge Router",
            "name": "Edge Router",
            "slug": "edge-router",
            "color": "9e9e9e",
            "vm_role": false,
            "description": "",
            "tags": [],
            "custom_fields": {},
            "created": "2022-11-09T15:04:19.210262Z",
            "last_updated": "2022-11-09T15:04:33.130616Z",
            "device_count": 3,
            "virtualmachine_count": 0
        },
```

I know this is likely something stupid I'm missing, but I tried entering the Device Role ID, Name and Slug all into the "NETBOX_DEVICE_ROLES" variable, but none seem to work. ☹ Any help would be appreciated.

**Comment by Paktosan** [comment]
Are you sure that `NETBOX_API` is set correct?
The Netbox API lives under `/api/` as you can also see at the `url` value of your role example.

**Comment by gmazoyer** [comment, marked as answer]
When using the docker setup, `NETBOX_DEVICE_ROLES` must be a single string composed of all roles separated by commas like `router,firewall,switch`. Don't worry about the URL missing the `/api/` it is automatically added by the lib dealing with API requests.

**Reply by ds2600** [reply]
Thank you!
